pkg/api/graph/graph.go

```diff
@@ -122,7 +122,7 @@ var DefaultNamer Namer = namer{}
 func (namer) ResourceName(obj interface{}) string {
 	switch t := obj.(type) {
 	case uniqueNamer:
-		return t.UniqueName()
+		return t.UniqueName().UniqueName()
```

Comment:
can you elaborate on why the signature change had to be made in the first place?  (I see you said something about reflection issues in the PR summary).